Issue 14348 - typeof(x).ident is not accepted as a symbol

yebblies opened the pull request:
Instead of evaluating as a type and then converting to a symbol, re-use the resolve logic to find the symbol.

https://issues.dlang.org/show_bug.cgi?id=14348

9rnsr commented on src/mtype.c:
```diff
@@ -6872,10 +6872,12 @@ Type *TypeTypeof::syntaxCopy()
+    Dsymbol *s;
+    resolve(loc, sc, &e, &t, &s);
+
+    return s;
```
Looks good. 

I noticed that same bug exists in `TypeReturn::toDsymbol()`. Could you also fix that?

```d
class C14348
{
    int foo() { return 1; }
}

C14348 foo14348()
{
    alias foo = typeof(return).foo;  // currently doesn't work.
    return null;
}
```

yebblies replied:
Added.

9rnsr replied:
Thanks!

yebblies committed:
Fix Issue 14348 - typeof(x).ident is not accepted as a symbol

Instead of evaluating as a type and then converting to a symbol, re-use the resolve logic to find the symbol.

9rnsr commented:
Auto-merge toggled on

yebblies commented:
Thanks!
